### ISSUE TITLE
Feature Request: `#[exclude = "example/some_big_directory"]`

### ISSUE THREAD

**Issue by NfNitLoop** [pyros2097/rust-embed#96]
I was just about to open a bug about how slow rust-embed was performing in release mode...

... then I found that I was accidentally embedding a 70Mb+ `node_modules` directory. Oops.  (Still, it did seem rather slow for a mere 70MB but maybe that's because it's a lot of small files?)

It would be nice if I could embed some directory, but exclude some large subdirectories from that embed.  Ex:

```
#[derive(RustEmbed)]
#[folder = "examples/"]
#[exclude = "examples/big_dir"]
```

Or, you could make it respect .gitignores? I already had that `node_modules` directory ignored, which is part of the reason I'd forgotten about it.

Otherwise, thanks for a great crate!

**Comment by pyros2097** [comment]
@NfNitLoop Yes the exclude folder would be needed in this case. Ignoring folders in .gitignore might be harder to implement for now. 
Anyone else here has any thoughts/ideas on how this should be done?
```rust
#[derive(RustEmbed)]
#[folder = "examples/", exclude = "big_dir"]
```

**Comment by paolobarbolini** [comment]
I think having the ability to exclude multiple folders would be enough for the moment. Having rust-embed follow .gitignore would probably cause more harm than good, since most projects git ignore the output of module bundlers like webpack but need those folders for the Rust code to work.

**Comment by Mcat12** [comment]
In order to exclude multiple folders, it may need to look like this:
```rust
#[derive(RustEmbed)]
#[folder = "examples/", exclude("big_dir1", "big_dir2")]
```

**Comment by Boscop** [comment]
If excluding folders gets allowed, what about individual (big) files, and what about globs? 🤔
Also, it would make sense to also use the shell-expand feature on the exclusion paths.

**Comment by qrnch-jan** [comment]
I'm looking for something that's mostly similar, but I have need for something more granular.  Basically there are two issues for us:

1) We have a master repo with several Rocket projects in it.  These Rocket projects use a shared "htstatic" directory in the tree which contains `img`, `css` and `js` with their respective files.  Some files within are used by all Rocket applications, some are not.  Currently rust-embed will add all files to all Rocket projects, unless we split the htstatic directory up, which we really don't want to do.
2) For debug builds we have our `foo.js` and our `foo.css` with all its bells and whistles, but for the release builds we want to use "minified" versions of them.  Our build system builds these into `foo.min.js` and `foo.min.css`, and in the final binary we want to include the ".min." versions, but not the originals.

In our case it would be really useful if:
The debug build just publishes everything, as it currently does.
But for the release build there should be a way to either exclude some entries _or_ to explicitly state what should be included.  In our case we'd prefer to have something like:

```
#[folder = "../../libs/htstatic/", include_manifest = "release_files"]
```

.. and `release_files` would be a file of file/directory entries that should be included in the list for release builds.

The upside of being able to specify an external manifest file, rather than adding the entries to the source code, is that build systems can generate these lists as appropriate for the platform it is building on.

Also, I'd like to voice some concerns about using `.gitignore` as an exclude list:  Not everyone uses git, and not all projects live in the root of a repository.  Please pick a name which is agnotic to the source management system.

**Comment by pyros2097** [comment]
This is implemented and released in v6.1.0
